handlers/delete_lease_test.go

```diff
@@ -63,13 +75,56 @@ func TestDeleteLeaseNoSuchLease(t *testing.T) {
 	nsListerDeleter := newFakeNamespaceListerDeleter(nil, nil, nil)
 	hdl := DeleteLease(getterUpdater, "claimer", nsListerDeleter)
 	req, err := http.NewRequest("DELETE", "/lease/"+uuid.New(), nil)
+	req.Header.Set("Authorization", "some awesome token")
 	assert.NoErr(t, err)
 	res := httptest.NewRecorder()
 	hdl.ServeHTTP(res, req)
 	assert.Equal(t, res.Code, http.StatusConflict, "response code")
 }
 
+func TestDeleteLeaseInvalidAuth(t *testing.T) {
+	os.Setenv("AUTH_TOKEN", "some awesome token")
+	defer os.Clearenv()
+	clusterNames := []string{"cluster1"}
+	uuids := make([]uuid.UUID, len(clusterNames))
+	annos := testutil.GetRawAnnotations(
+		clusterNames,
+		leases.TimeFormat,
+		func(i int) time.Time {
+			return time.Now().Add(1 * time.Hour)
+		},
+		func(i int) uuid.UUID {
+			ret := uuid.NewUUID()
+			uuids[i] = ret
+			return ret
+		},
+	)
+	getterUpdater := newFakeServiceGetterUpdater(
+		&api.Service{ObjectMeta: api.ObjectMeta{Annotations: annos}},
+		nil,
+		nil,
+		nil,
+	)
+	path := "/lease/" + uuids[0].String()
+	namespaces := []string{"ns1", "ns2"}
+	nsList := api.NamespaceList{}
+	for _, namespace := range namespaces {
+		nsList.Items = append(nsList.Items, api.Namespace{ObjectMeta: api.ObjectMeta{Name: namespace}})
+	}
+	nsListerDeleter := newFakeNamespaceListerDeleter(
+		&nsList, nil, nil)
+	hdl := DeleteLease(getterUpdater, "claimer", nsListerDeleter)
```

Comment:
since the `DeleteLease` handler no longer checks auth (it should be done with the `WithAuth` middleware), this test functionality would be better suited as a unit test against that `WithAuth` middleware